source/diff/diff.cpp

```diff
@@ -338,6 +338,30 @@ class Differ {
       std::function<void(const IdGroup& src_group, const IdGroup& dst_group)>
           match_group);
 
+  // Bucket `src_ids` and `dst_ids` by key ids returned by `get_group`, and
+  // then pair up buckets whose ids map to each other.
+  //
+  // For example, say the input groups are as follows, where `X -> Y` means
+  // that `X` is a member of the `IdGroup` and `get_group` maps `X` to `Y`
+  //
+  // - src_ids: { 1 -> 100, 2 -> 300, 3 -> 100, 4 -> 200 }
+  // - dst_ids: { 5 -> 10, 6 -> 20, 7 -> 10, 8 -> 300 }
+  //
+  // and suppose that the following src/dst ids are paired: 100/10 and 200/20.
```

Comment:
nit: To keep with the terminology used in this file, `s/are paired/are already matched`. It's been a while since I worked on this code, so please ignore if I'd used this terminology elsewhere already.